qcfractal/storage_sockets/mongoengine_socket.py

```diff
@@ -1041,18 +1042,18 @@ def get_results(self,
         meta = storage_utils.get_metadata()
         query = {}
         parsed_query = {}
-        if program:
+        if program is not None:
             query['program'] = program
-        if method:
+        if method is not None:
             query['method'] = method
-        if basis:
-            query['basis'] = basis
-        if molecule:
+        if basis is not None:
+            query['basis'] = basis if basis != 'null' else None
+        if molecule is not None:
             query['molecule'], _ = _str_to_indices_with_errors(molecule)
-        if driver:
+        if driver is not None:
             query['driver'] = driver
-        if keywords:
-            query['keywords'] = keywords
+        if keywords is not None:
+            query['keywords'] = keywords if keywords != 'null' else None
```

Comment:
I think this should be `keywords if keywords != '' else None` now?

Reply:
Yes of course

Reply:
I think we should remove this now since we already have the default: `""`

Reply:
+1 for that.